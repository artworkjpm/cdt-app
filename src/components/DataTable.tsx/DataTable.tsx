interface Props {
	tableHeaders: string[];
	data: any[];
}

export default function DataTable({ tableHeaders, data }: Props) {
	console.log(tableHeaders, data);

	return (
		<div>
			<table>
				<thead>
					<tr>
						{tableHeaders.map((item, index) => {
							return <th key={index}>{item}</th>;
						})}
					</tr>
				</thead>
				<tbody>
					{data.map((item, index) => {
						return (
							<tr key={index}>
								{tableHeaders.map((el, i) => {
									return <td>{item[el.toLowerCase()]}</td>;
								})}
							</tr>
						);
					})}
				</tbody>
			</table>
		</div>
	);
}
