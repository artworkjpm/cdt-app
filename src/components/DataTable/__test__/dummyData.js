export const dummyData = [
	{
		id: 20,
		name: "20th Design",
		courses: 111,
		wales: 333,
		last_updated: "12/04/2021",
		by: "WD"
	},
];
