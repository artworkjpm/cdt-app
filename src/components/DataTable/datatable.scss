@import "../../styles/variables.scss";

.initials {
	border-radius: 50%;
	width: 50px;
	height: 50px;
	background: $mainColor;
	line-height: 3.2;
	color: white;
	font-weight: bold;
	margin: 0 auto;
}
