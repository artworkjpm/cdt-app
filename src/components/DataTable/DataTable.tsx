import { useState } from "react";
import DialogEdit from "../Dialog/Dialog";
import "./datatable.scss";

interface Props {
	tableHeaders: string[];
	data: any[];
	editTitle: string;
	editableFields: string[];
	onEditSubmit: any;
	openDialog: any;
	handleClose: any;
	handleOpen: any;
}

export default function DataTable({ tableHeaders, data, editTitle, editableFields, onEditSubmit, openDialog, handleClose, handleOpen }: Props) {
	const [editData, setEditData] = useState();

	const handleClickOpen = (itemData: any) => {
		handleOpen();
		setEditData(itemData);
	};

	return (
		<div id="datatable">
			<table>
				<thead>
					<tr>
						{tableHeaders.map((item, index) => {
							return <th key={index}>{item.replace("_", " ")}</th>;
						})}
					</tr>
				</thead>
				<tbody>
					{data.map((item, index) => {
						return (
							<tr id="clicker" key={index} onClick={() => handleClickOpen(item)}>
								{tableHeaders.map((el, i) => {
									return <td key={i}>{item[el.toLowerCase()]}</td>;
								})}
							</tr>
						);
					})}
				</tbody>
			</table>
			{openDialog && <DialogEdit title={editTitle} openDialog={openDialog} handleClose={handleClose} editData={editData} editableFields={editableFields} onEditSubmit={onEditSubmit} />}
		</div>
	);
}
