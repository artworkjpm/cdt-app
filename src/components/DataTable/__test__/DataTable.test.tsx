import React from "react";
import { shallow } from "enzyme";
import DataTable from "../DataTable";
import { dummyData } from "./dummyData";

import renderer from "react-test-renderer";

describe("<DataTable/>", () => {
	test("should have specific props", () => {
		let props = { tableHeaders: ["Name", "Courses", "Wales", "Last_Updated", "By"], data: dummyData, editTitle: "Designs", editableFields: ["name", "courses", "wales"], onEditSubmit: "", openDialog: "", handleClose: "", handleOpen: "" };

		const fakeDataTable = renderer.create(<DataTable {...props} />).toJSON();
		expect(fakeDataTable).toMatchSnapshot();
	});
});
