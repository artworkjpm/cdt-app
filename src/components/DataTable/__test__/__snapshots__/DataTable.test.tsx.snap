// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTable/> should have specific props 1`] = `
<div
  id="datatable"
>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Courses
        </th>
        <th>
          Wales
        </th>
        <th>
          Last Updated
        </th>
        <th>
          By
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        id="clicker"
        onClick={[Function]}
      >
        <td>
          20th Design
        </td>
        <td>
          111
        </td>
        <td>
          333
        </td>
        <td>
          12/04/2021
        </td>
        <td>
          WD
        </td>
      </tr>
    </tbody>
  </table>
  
</div>
`;
